### PR TITLE
fix to stop assessment from throwing errors in IE8

### PR DESCRIPTION
**Pull request by moloko**
as per https://github.com/adaptlearning/adapt_framework/issues/1432

### CODE REVIEW

File: js/adapt-assessmentArticleModel.js [hunk 1]
```diff
@@ -666,24 +671,24 @@ define([
...
             var questions = [];
-            for (var id in indexByIdQuestions) {
-                if (Adapt.findById(id).get("_isQuestionType")) {
-                    questions.push({
-                        _id: id,
-                        _isCorrect: indexByIdQuestions[id]
-                    });
+            for (id in indexByIdQuestions) {
+                if(indexByIdQuestions.hasOwnProperty(id)) {
+                    if (Adapt.findById(id).get("_isQuestionType")) {
```
Comment (tomgreenfield) [inline js/adapt-assessmentArticleModel.js]: Doesn't need to stop you merging but I'd avoid the double nested if statements by concatanating the conditions with `&&`.

**Comment by moloko** [comment]
thanks @tomgreenfield - have amended that and also added an explanatory comment

anyone want to review again?

**Comment by moloko** [comment]
OK, anyone have any objection if I go ahead and merge?

**Comment by tomgreenfield** [comment]
No objections.

**Pull request merged**